### PR TITLE
Wait one hour to sweep StreamStore Value table

### PR DESCRIPTION
**Pull request by fsamuel-bs**
**Goals (and why)**: Wait one hour after sweeping StreamStore Index table before Sweeping StreamStore Value table.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2797)

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/StreamStoreRemappingNextTableToSweepProviderImpl.java [hunk 1]
```diff
@@ -17,39 +17,62 @@
...
 import com.palantir.atlasdb.schema.stream.StreamTableType;
 import com.palantir.atlasdb.transaction.api.Transaction;
 
 public class StreamStoreRemappingNextTableToSweepProviderImpl implements NextTableToSweepProvider {
+    private static final long MILLIS_TO_WAIT_BEFORE_SWEEP_VALUE_TABLE = TimeUnit.HOURS.toMillis(1);
+
     private NextTableToSweepProviderImpl delegate;
-    private boolean hasRemappedStreamStoreValueTable = false;
-    private TableReference previousStreamStoreValueTable = null;
+    private State state = State.DEFAULT;
```
Comment (nziebart) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/StreamStoreRemappingNextTableToSweepProviderImpl.java]: does this logic need to be stateful? seems like you could do:
- if last values table sweep time > last index table sweep time:
   values priority = 0
   index priority = priority we'd normally give the values table
- else if last index table sweep < 2 hours ago:
   values priority = 0
   index priority = 0
- else (last index sweep > 2 hours ago and last values sweep < last index sweep)
   values priority = normal
   index priority = 0

Comment (nziebart) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/StreamStoreRemappingNextTableToSweepProviderImpl.java]: just saw that @hsaraogi suggested a similar thing as well

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/StreamStoreRemappingNextTableToSweepProviderImpl.java [hunk 2]
```diff
@@ -16,40 +16,96 @@
...
+
+    private boolean didSweepAllIndexAndValueTables(
+            Map<TableReference, SweepPriority> indexToPriority,
+            Map<TableReference, SweepPriority> valueToPriority) {
+        return indexToPriority.size() == valueToPriority.size() && indexToPriority.size() != 0;
+    }
+
+    // if ss.value >= ss.index -> ignore value and sweep index
+    // if ss.value <  ss.index && ss.index <= 1 hour ago -> ignore value
+    // if ss.value <  ss.index && ss.index >  1 hour ago -> sweep value
```
Comment (fsamuel-bs) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/StreamStoreRemappingNextTableToSweepProviderImpl.java]: This should be a Javadoc if the code below is not clear enough.

**Comment by tboam** [comment]
Closing and moving to the other branch to review in one go.  Functionality from both PRs has been refactored.

**Pull request merged**